pandas/tests/arrays/test_datetimelike.py

```diff
@@ -285,10 +303,10 @@ def test_array_i8_dtype(self, tz_naive_fixture):
         result = np.array(arr, dtype=np.int64)
         tm.assert_numpy_array_equal(result, expected)
 
-        # check that we are not making copies when setting copy=False
+        # check that we are still making copies when setting copy=False
```

Comment:
what changed here?

Reply:
This is as a follow-up to https://github.com/pandas-dev/pandas/pull/23524 that added it "wrongly" (as I commented there as well). 
And with "wrongly", I mean not following numpy semantics. Numpy always copies if the dtype is not exactly equal, even if no copy would be necessary. So doing `np.asarray(M8_array, dtype='int64')` still copies the data. I decided to follow that logic here as well (by not implementing it, and letting numpy do the work).